net/download/service/src/main/java/de/danoeh/antennapod/net/download/service/episode/autodownload/EpisodeCleanupAlgorithmFactory.java

```diff
@@ -4,9 +4,6 @@
 
 public abstract class EpisodeCleanupAlgorithmFactory {
     public static EpisodeCleanupAlgorithm build() {
-        if (!UserPreferences.isEnableAutodownload()) {
-            return new APNullCleanupAlgorithm();
-        }
```

Comment:
We need to be careful with the default here. It looks like this means that AntennaPod will start deleting episodes after the upgrade, even if you don't use auto-download. If auto-download is disabled globally, we should change the default to `UserPreferences.EPISODE_CLEANUP_NULL` when upgrading (`PreferenceUpgrader`) and make sure that the default for new installs is also `UserPreferences.EPISODE_CLEANUP_NULL`

Reply:
As far as i understand the cleanup code, the default for new installations should already be "null".

I have added the upgrades, but this probably needs to be tested to really make sure it does what you intend it to do.